app/controllers/complaints_controller.rb

```diff
@@ -154,6 +154,35 @@ def update_status
     redirect_to complaint_path(@complaint.access_token)
   end
 
+  def change_content_type
+    unless current_user&.same_as?(@complaint.assignee)
+      flash[:danger] = 'You are not assigned to this report. Assign yourself before changing the content type.'
+      redirect_back fallback_location: complaint_path(@complaint.access_token) and return
+    end
+
+    new_content_type = helpers.content_type(params[:new_content_type])
+    if new_content_type.nil?
+      flash[:danger] = 'Invalid content type.'
+      redirect_back fallback_location: complaint_path(@complaint.access_token) and return
+    end
+
+    update_params = { content_type: params[:new_content_type] }
+    message = "Content type changed to #{new_content_type['name']} by #{current_user.username}."
+    @comment = @complaint.comments.new(user_id: -1, internal: true, content: message)
```

Comment:
FYI, @Oaphi just added a helper to get the system user, to avoid hard-coding constants.